### PR TITLE
JAMES-2187 Download email and attachment work on a delegated message

### PR DESCRIPTION
**Pull request by quynhn**

### CODE REVIEW

File: server/protocols/jmap-integration-testing/jmap-integration-testing-common/src/test/resources/cucumber/DownloadGet.feature [hunk 1]
```diff
@@ -61,4 +62,13 @@ Feature: Download GET
     And a connected user "username1@domain.tld"
     And "username1@domain.tld" has a mailbox "INBOX"
     When "username1@domain.tld" downloads "1"
-    Then the user should receive a not found response
\ No newline at end of file
+    Then the user should receive a not found response
+
+  Scenario: User cannot download attachment of another user when shared mailbox
```
Comment (rouazana) [inline server/protocols/jmap-integration-testing/jmap-integration-testing-common/src/test/resources/cucumber/DownloadGet.feature]: s/cannot/can/ ?

File: server/protocols/jmap-integration-testing/jmap-integration-testing-common/src/test/resources/cucumber/DownloadGet.feature [hunk 2]
```diff
@@ -61,4 +62,13 @@ Feature: Download GET
...
+    Given "username@domain.tld" mailbox "sharedMailbox" contains a message "1" with an attachment "2"
+    And "username@domain.tld" shares its mailbox "sharedMailbox" with "username1@domain.tld"
+    And a connected user "username1@domain.tld"
+    And "username1@domain.tld" has a mailbox "sharedMailbox"
+    When "username1@domain.tld" downloads "2"
+    Then the user should receive that blob
+    And the blob size is 3071
```
Comment (rouazana) [inline server/protocols/jmap-integration-testing/jmap-integration-testing-common/src/test/resources/cucumber/DownloadGet.feature]: what about adding tests about downloading message content also? (here you only test downloading attachments)

**Comment by mbaechler** [comment]
test this please

File: server/protocols/jmap-integration-testing/jmap-integration-testing-common/src/test/resources/cucumber/DownloadGet.feature (hunk 1 above)
Comment (strokyl) [reply server/protocols/jmap-integration-testing/jmap-integration-testing-common/src/test/resources/cucumber/DownloadGet.feature]: :+1

**Comment by strokyl** [comment]
This PR has been rebased on top of this one: https://github.com/linagora/james-project/pull/1040

**Pull request closed without merging**